src/expressions.md

```diff
@@ -133,96 +136,88 @@ assert_eq!(
 );
 ```
 
-> **Note**: Since this is applied recursively, these expressions are also
-> evaluated from innermost to outermost, ignoring siblings until there are no
-> inner subexpressions.
+> **Note**: Since this is applied recursively, these expressions are also evaluated from innermost to outermost, ignoring siblings until there are no inner subexpressions.
 
-## Place Expressions and Value Expressions
+## Expression categories
 
-Expressions are divided into two main categories: place expressions and
-value expressions. Likewise within each expression, operands may occur
-in either place context or value context. The evaluation of an expression
-depends both on its own category and the context it occurs within.
+<!-- This section was renamed, but heavily linked to. So adding this span to keep the links working. -->
+<span id="place-expressions-and-value-expressions"></span>
 
-A *place expression* is an expression that represents a memory location. These
-expressions are [paths] which refer to local variables, [static variables],
-[dereferences][deref] (`*expr`), [array indexing] expressions (`expr[expr]`),
-[field] references (`expr.f`) and parenthesized place expressions. All other
-expressions are value expressions.
+Expressions and expression contexts are divided into two *expression categories*: *place* and *value*.
+For expressions, what they evaluate into determines their categories.
```

Comment:
This feels awkward. We already established the "place or value" dichotomy earlier, so IMO it'd be better to say upfront (L45) that there are "place expressions" and "value expressions" -- or we find some way to not make the place/value distinction at all until here.

Reply:
My thought was that L45 gives a summary, and then this section expands upon the idea. It would feel weird to not include the dichotomy in the main definition but including the terms in the main definition would be more detail then necessary.

Also, the fact that the category exists for both the expression and the expression context makes this harder to describe.

Reply:
> Also, the fact that the category exists for both the expression and the expression context makes this harder to describe.

These are not two independent facts, so maybe there is a way to present them as one fact? There are two kinds of expressions, basically two kinds of "non-terminals" in the grammar (`ValExpr`, `PlaceExpr`), so naturally there are two questions: which class does eache xpression fall in, and which `Expr` in the grammar is refined to each of these two non-terminals?